src/uct/tcp/tcp_sockcm_ep.c

```diff
@@ -327,28 +330,20 @@ ucs_status_t uct_tcp_sockcm_ep_server_handle_data_received(uct_tcp_sockcm_ep_t *
 {
     uct_tcp_sockcm_priv_data_hdr_t *hdr = (uct_tcp_sockcm_priv_data_hdr_t *)
                                            cep->comm_ctx.buf;
+    uint16_t client_state;
     ucs_status_t status;
     int events;
 
-    if (cep->state == (UCT_TCP_SOCKCM_EP_ON_SERVER |
-                       UCT_TCP_SOCKCM_EP_DATA_RECEIVED)) {
-        status = uct_tcp_sockcm_ep_server_invoke_conn_req_cb(cep);
-        if (status != UCS_OK) {
-            goto out;
-        }
-
-        /* server to send its private data to the client and listen to disconnect events */
-        events = UCS_EVENT_SET_EVWRITE | UCS_EVENT_SET_EVREAD;
-    } else if (cep->state == (UCT_TCP_SOCKCM_EP_SERVER_CONN_CREATED |
-                              UCT_TCP_SOCKCM_EP_DATA_RECEIVED       |
-                              UCT_TCP_SOCKCM_EP_CONN_SENT)) {
+    if (ucs_test_all_flags(cep->state, UCT_TCP_SOCKCM_EP_SERVER_CONN_CREATED |
+                                       UCT_TCP_SOCKCM_EP_DATA_RECEIVED       |
+                                       UCT_TCP_SOCKCM_EP_CONN_SENT)) {
```

Comment:
is it possible that CONN_SENT is true, but UCT_TCP_SOCKCM_EP_SERVER_CONN_CREATED or UCT_TCP_SOCKCM_EP_DATA_RECEIVED are false?

Reply:
it should not be possible but we check here to make sure the flow is correct and there is no error.

Reply:
so pls convert it to assertion